### PR TITLE
Update codeowners file to match runtime repo

### PR DESCRIPTION
**Pull request by gewarren**
Updating based on https://github.com/dotnet/runtime/blob/master/docs/area-owners.md.
Also replaces `**/**` syntax that seems to be invalid.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (carlossanlop) [inline .github/CODEOWNERS:21]: Does the suffix `*/` also include any files inside that folder, as opposed to using `*/*`?

And I know we don't have any folders inside `xml/Namespace/`, only xmls, so it makes sense to not use `**`.

File: .github/CODEOWNERS
Comment (eiriktsarpalis) [inline .github/CODEOWNERS:22]: this would need to be updated every time area-owners.md is changed. Any chance we could consolidate into one document? (e.g. by adding a column linking to the relevant docs folders in area-owners.md)

cc @carlossanlop @jeffhandley

File: .github/CODEOWNERS
Comment (gewarren) [reply .github/CODEOWNERS:21]: Yes, I believe `*/` includes files inside that folder (including in subdirectories, according to [this](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax)).

We think it's invalid because it's not currently functional in requesting PR reviews. And the link above states:

"If any line in your CODEOWNERS file contains invalid syntax, the file will not be detected and will not be used to request reviews."

File: .github/CODEOWNERS
Comment (gewarren) [reply .github/CODEOWNERS:22]: I will leave it up to you and Carlos to decide how reviewers are tagged. If we decide to rely on fabricbot, I think we should just delete this file.

**Review by carlossanlop — APPROVED** [review]

**Comment by gewarren** [comment]
Looks like it's working correctly now. Yippee!

https://github.com/dotnet/dotnet-api-docs/pull/5390

image

**Pull request merged**
